RxCocoa/Common/DelegateProxyType.swift

```diff
@@ -144,41 +181,42 @@ extension DelegateProxyType {
     ///
     ///     extension Reactive where Base: UISearchBar {
     ///
-    ///         public var delegate: DelegateProxy {
-    ///            return RxSearchBarDelegateProxy.proxyForObject(base)
+    ///         public var delegate: DelegateProxy<Base, UISearchBarDelegate> {
+    ///            return RxSearchBarDelegateProxy<Base>.proxy(for: base)
     ///         }
     ///
     ///         public var text: ControlProperty<String> {
     ///             let source: Observable<String> = self.delegate.observe(#selector(UISearchBarDelegate.searchBar(_:textDidChange:)))
     ///             ...
     ///         }
     ///     }
-    public static func proxyForObject(_ object: AnyObject) -> Self {
+    public static func proxy(for object: ParentObject) -> Self {
         MainScheduler.ensureExecutingOnScheduler()
 
-        let maybeProxy = Self.assignedProxyFor(object) as? Self
+        let maybeProxy = self.assignedProxy(for: object)
 
-        let proxy: Self
+        // Type is ideally be `(Self & Delegate)`, but Swift 3.0 doesn't support it.
+        let proxy: Delegate
         if let existingProxy = maybeProxy {
             proxy = existingProxy
         }
         else {
-            proxy = Self.createProxy(for: object) as! Self
-            Self.assignProxy(proxy, toObject: object)
-            assert(Self.assignedProxyFor(object) === proxy)
+            proxy = castOrFatalError(self.createProxy(for: object))
+            self.assignProxy(proxy, toObject: object)
+            assert(self.assignedProxy(for: object) === proxy)
         }
-
-        let currentDelegate: AnyObject? = Self.currentDelegateFor(object)
-
-        if currentDelegate !== proxy {
-            proxy.setForwardToDelegate(currentDelegate, retainDelegate: false)
-            assert(proxy.forwardToDelegate() === currentDelegate)
-            Self.setCurrentDelegate(proxy, toObject: object)
-            assert(Self.currentDelegateFor(object) === proxy)
-            assert(proxy.forwardToDelegate() === currentDelegate)
+        let currentDelegate = self.currentDelegate(for: object)
+        let delegateProxy = unsafeDowncast(proxy, to: self)
```

Comment:
The comment thread was closed by github, but still problem is here.

This is to solve cast between `RxTableViewDelegateProxy<UITableView>` and `RxScrollViewDelegateProxy<UIScrollView>`.
But I think it's better that quit using it. 
Remove `ParentObject` type generics parameter from `RxScrollViewDelegateProxy`, then we don't need to use `unsafeDowncast` anymore.